src/object.c

```diff
@@ -149,26 +149,31 @@ robj *tryCreateStringObject(const char *ptr, size_t len) {
  * integer, because the object is going to be used as value in the Redis key
  * space (for instance when the INCR command is used), so we want LFU/LRU
  * values specific for each key. */
-robj *createStringObjectFromLongLongWithOptions(long long value, int valueobj) {
+#define LL2STROBJ_AUTO 0
+#define LL2STROBJ_VALUE 1
+#define LL2STROBJ_EMB 2
```

Comment:
following the previous comment (about caller use case), i think these should be the low level flags.
```suggestion
#define LL2STROBJ_AUTO 0
#define LL2STROBJ_NO_INT_ENC 1 /* disallow integer encoded objects. */
#define LL2STROBJ_NO_SHARED 2 /* disallow shared objects */
```
and then, i'd move the part that checks the maxmemory_policy to the outer function (createStringObjectFromLongLongForValue) to control the input flags for this one.

p.s. some comments about LRU should be moved along too.

Reply:
Thanks for your nice suggestions, I like it. After these alternations, I noticed that `tryObjectEncodingEx` have a similar implementation for interger object, do you think we should keep trying to unify these codes?
https://github.com/redis/redis/blob/cefe4566e93e07c4c636286d7f2d5a3d4b9a7b24/src/object.c#L658-L679

Reply:
i think that's a bit more complicated, i think we better leave that aside for now.